### PR TITLE
 JAMES-2544 Store mime message as blobs for enqueued mail 

### PR DESCRIPTION
**Pull request by aduprat**
Only last commit has to be reviewed in this PR: 3a431c7

### CODE REVIEW

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java [hunk 1]
```diff
@@ -420,6 +424,22 @@ default void browseShouldNotFailWhenConcurrentRemoveWhenIterating() throws Excep
...
+        mailQueue.enQueue(defaultMail()
+            .name("mail with blob")
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .setSubject("mail subject")
+                .setText("mail body")
+                .build())
+            .build());
+
+        MimeMessage mimeMessage = mailQueue.browse().next().getMail().getMessage();
+        assertAll(
```
Comment (mbaechler) [inline server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java]: softly assert ?

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java [hunk 2]
```diff
@@ -420,6 +424,22 @@ default void browseShouldNotFailWhenConcurrentRemoveWhenIterating() throws Excep
         assertThatCode(() ->  Iterators.consumingIterator(items)).doesNotThrowAnyException();
     }
 
+    @Test
+    default void browseShouldReturnMailsWithMimeMessage() throws Exception {
```
Comment (mbaechler) [inline server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java]: I thought we decided exactly the opposite, did I missed something ?

**Comment by chibenwa** [comment]
Forced pushed to solve conflict

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java (hunk 2 above)
Comment (chibenwa) [reply server/queue/queue-api/src/test/java/org/apache/james/queue/api/ManageableMailQueueContract.java]: Browsing needs the mime-message (or the admin is blind) but some operations like updating browse start / removes do not need it.

**Comment by aduprat** [comment]
:+1:

File: server/queue/queue-file/src/test/java/org/apache/james/queue/file/FileMailQueueTest.java [hunk 3]
```diff
@@ -97,4 +97,11 @@ public void removeBySenderShouldNotFailWhenQueueIsEmpty() {
     public void removeByRecipientShouldRemoveSpecificEmailWhenMultipleRecipients() {
 
     }
+
+    @Test
+    @Override
+    @Disabled("JAMES-2544 Not supported yet")
+    public void browseShouldReturnMailsWithMimeMessage() {
```
Comment (mbaechler) [inline server/queue/queue-file/src/test/java/org/apache/james/queue/file/FileMailQueueTest.java]: browseShouldReturnMailsWithoutMimeMessage

**Comment by aduprat** [comment]
conflict: `push -f`

**Comment by aduprat** [comment]
```
[ERROR] Errors: 
[ERROR]   RabbitMQClusterTest$ClusterNodesFailure.connectingToAClusterWithAFailedRabbit » Completion
```
test this please

**Comment by aduprat** [comment]
```
[ERROR] Errors: 
[ERROR]   RabbitMQClusterTest$ClusterSharing.rabbitMQManagerShouldReturnThreeNodesWhenAskingForStatus » Completion
```
test this please

**Comment by aduprat** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**